tests/integration/dhcp_test.py

```diff
@@ -702,54 +711,79 @@ def _get_running_routes():
     return libnmstate.show().get(Constants.ROUTES, {}).get(RT.RUNNING, [])
 
 
+def _retry_till_true(func):
```

Comment:
I suggest to use a decorator for this and place it under testlib.

Example: https://github.com/oVirt/vdsm/blob/master/tests/network/functional/netfunctestlib.py#L88

If needed, we can add to it a timeout argument as an extension.